src/evm_context.rs

```diff
@@ -174,7 +174,16 @@ pub fn basefee(ssa: &mut SSATracker) -> SMTVariable {
 pub fn calldataload(index: SMTExpr, ssa: &mut SSATracker) -> SMTExpr {
     let calldata = context().calldata.smt_var(ssa);
     let arguments = (0..32)
-        .map(|i| smt::uf(calldata.clone(), vec![smt::bvadd(index.clone(), i)]))
+        .map(|i| {
+            // TODO this could benefit from let expressions.
+            // We are repeating the expression that computes the index.
+            let sub_index = smt::bvadd(index.clone(), i);
```

Comment:
Should we do something about the overflow now or later?

Reply:
I don't see what we could do - to prevent overflow we would have to make this an integer addition, i.e. make the calldata function a function from the integers and that would mean a bv2uint for each access...

Reply:
or we could just add a real life assumption that `calldatasize() + 32 < 2^256`